am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/RevokeAccessIntegrationTest.java

```diff
@@ -22,17 +29,21 @@ void setupTest() {
     }
 
     @Test
-    void revokeResourceAccess_whenRevokingResourceAccess_ResourceAccessRemovedFromDatabase() {
-        ams.createResourceAccess(createRecord(resourceId, ACCESSOR_ID, EXPLICIT_READ_CREATE_UPDATE_PERMISSIONS));
+    void whenRevokingResourceAccessResourceAccessRemovedFromDatabase() throws TransactionRolledbackException {
+        Map<JsonPointer, Set<Permission>> singleAttributePermission = new ConcurrentHashMap<>();
+        singleAttributePermission.put(JsonPointer.valueOf(""), EXPLICIT_READ_CREATE_UPDATE_PERMISSIONS);
+
+
+        ams.grantExplicitResourceAccess(grantAccess(resourceId, ACCESSOR_ID, singleAttributePermission));
```

Comment:
I am not sure refactoring of that size was needed. You could have created overloaded method `createAccessGrantForWholeDocument` which accepts only sets of permissions. Attribute "" would be hidden within factory method.  It would only require change from `createRecord` to `createAccessGrantForWholeDocument`.

This way we would avoid nastiness of:

```
Map<JsonPointer, Set<Permission>> singleAttributePermission = new ConcurrentHashMap<>();
         singleAttributePermission.put(
```